File: Q07/main.cpp
#include<iostream>
using namespace std;
int main(){
    int n;
    cin>>n;
    int arr[n];
    for(int i=0;i<n;i++){
        cin>>arr[i];
    }
    int start=0;
    for(int i=0;i<n;i++){
        if(arr[i]!=0){
            int temp=arr[i];
            arr[i]=arr[start];
            arr[start]=temp;
        }
    }
    for(int i=0;i<n;i++){
        cout<<arr[i];
    }
    return 0;
}

File: Q07/main.py
n=int(input())
arr=[]
for i in range(n):
    arr.append(int(input()))
start=0
for i in range(n):
    if arr[i]!=0:
        temp=arr[i]
        arr[i]=arr[start]
        arr[start]=temp
        start+=1
for i in range(n):
    print(arr[i],end=" ")